en/docs/develop/manage-repository/connect-your-own-github-repository-to-choreo.md

```diff
@@ -1,148 +1,69 @@
-
-    ![Authorize GitHub app](../../assets/img/tutorials/connect-own-repo/authorize-github-app.png){.cInlineImage-half}
-
-    !!!info
-
-         - The **Choreo GitHub App** requires the following permission:
-            - Read and write access to code and pull requests.
-            - Read access to issues and metadata.
-         - You can [revoke access](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/reviewing-your-authorized-integrations#reviewing-your-authorized-github-apps) if you do not want Choreo to have access to your GitHub account. However, write access is only used to send pull requests to a user repository. Choreo will not directly push any changes to a repository.
-
-7. Select a GitHub account and a GitHub repository that includes a **Choreo GitHub App** installation.
-
-    !!! tip 
-        Choreo does not allow you to connect GitHub repositories that include existing Ballerina projects.
+3. Click **Next**.
+4. Provide authorization with your GitHub account by clicking  **Authorize with GitHub**. 
+5. If you have not already authorized Choreo apps, click **Authorize Choreo Apps** when prompted.
+6. Select a GitHub account, a repository that includes the implementation, the relevant branch, and the build preset: Ballerina or Dockerfile. 
+7. Enter a valid path relative to the root of your repository that points to the implementation of the REST API. If you have not designed and implemented your REST API yet, you can connect an empty repo or a sub-folder and proceed to create the component. 
 
     !!! info
-        If a selected repository is already integrated with Choreo to create a component, you cannot reuse it to create another Choreo component.
-
-8. Click **Create** to proceed with component initialization. This displays a pull request similar to the following:
-   ![View pull request](../../assets/img/tutorials/connect-own-repo/view-pull-request.png){.cInlineImage-full}
-
-9. Click **View Pull Request**.
-10. Review and click **Merge pull request**, and then click **Confirm Merge**. When you merge the pull request, it adds the necessary metadata files to connect your GitHub repository to Choreo so that you can proceed to create the component.
-
-11. To configure the pre-commit hook that Choreo uses to extract configurables defined in the Ballerina code, follow the steps given below:
-
-      !!! info
-          When you develop a Choreo component via the Web Editor, Choreo configures this pre-commit hook automatically. However, if you prefer to develop the component locally, you need to configure this pre-commit hook manually.
-
-     1. Carry out a GitHub pull locally to update your clone with the latest changes.
-
-     2. In your terminal, navigate to the root directory of your GitHub repository and issue the following commands:
+        You can create a component by connecting an empty GitHub repository. But you can only deploy it after you implement it. 
```

Comment:
I think we need to rephrase this sentence `But you can only deploy it after you implement it.` like `But you must push a valid Ballerina REST API implementation (*.bal and Ballerina.toml) to make the component deployable.`